### PR TITLE
Google Calendar: Add google calendar block fixtures and tests

### PR DESCRIPTION
**Pull request by aaronrobertshaw**
#### Changes proposed in this Pull Request:
* Refactors Google Calendar block to functional component
* Add fixtures and block tests for Google Calendar block

#### Does this pull request change what data or activity we track or use?
No change.

#### Testing instructions:
1. Create a new post, adding a Google Calendar block to it including embed url. Save post.
2. Checkout this PR, reload editor ensuring block displays correctly in editor and frontend and works as expected
3. Run tests for the block `cd projects/plugins/jetpack && yarn jest extensions/blocks/google-calendar/test/`

### CODE REVIEW

File: projects/plugins/jetpack/extensions/blocks/google-calendar/controls.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+			<form onSubmit={ onSubmit } className={ className }>
+				<textarea
+					type="text"
+					value={ embedValue }
+					className="components-placeholder__input"
+					aria-label={ __( 'Google Calendar URL or iframe', 'jetpack' ) }
+					placeholder={ __( 'Enter URL or iframe to embed here…', 'jetpack' ) }
+					onChange={ onChange }
+				/>
+				<Button isSecondary isLarge type="submit">
```
Comment (andrewserong) [inline projects/plugins/jetpack/extensions/blocks/google-calendar/controls.js:21]: The `isLarge` attribute was already present before the refactor, but I can see it's throwing a warning when running in dev mode. Shall we remove it here?

image

File: projects/plugins/jetpack/extensions/blocks/google-calendar/edit.js [hunk 2]
```diff
@@ -19,183 +18,163 @@ import { getBlockDefaultClassName } from '@wordpress/blocks';
...
+		<form onSubmit={ handleEmbed } className={ formClassName }>
+			<textarea
+				type="text"
+				value={ editedEmbed }
+				className="components-placeholder__input"
+				aria-label={ __( 'Google Calendar URL or iframe', 'jetpack' ) }
+				placeholder={ __( 'Enter URL or iframe to embed here…', 'jetpack' ) }
+				onChange={ event => setEditedEmbed( event.target.value ) }
+			/>
+			<Button isSecondary isLarge type="submit">
```
Comment (andrewserong) [inline projects/plugins/jetpack/extensions/blocks/google-calendar/edit.js:97]: Another `isLarge` attribute that we can remove.

File: projects/plugins/jetpack/extensions/blocks/google-calendar/test/validate.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
+/**
+ * Internal dependencies
+ */
+import { name, settings } from '../';
+import runBlockFixtureTests from '../../../shared/test/block-fixtures';
+
+/**
+ * 👀 Update this array of blocks to contain the name and settings for **ALL**
```
Comment (andrewserong) [inline projects/plugins/jetpack/extensions/blocks/google-calendar/test/validate.js:8]: It doesn't hurt to leave these comments here, but shall we remove them to keep it tidy?

**Review by andrewserong — DISMISSED** [review]
Nice refactor @aaronrobertshaw! This is testing well, and it's looking much cleaner. Good and thorough tests, too 👍

Just left an optional comment about removing the `isLarge` props for the Button components since we don't need to include them anymore, but other than that, it's looking good to me! (And is testing well manually in the editor)

File: projects/plugins/jetpack/extensions/blocks/google-calendar/test/validate.js (hunk 3 above)
Comment (aaronrobertshaw) [reply projects/plugins/jetpack/extensions/blocks/google-calendar/test/validate.js:8]: That was an oversight on my part. I'll clean it up.

Comment (aaronrobertshaw) [reply projects/plugins/jetpack/extensions/blocks/google-calendar/test/validate.js:8]: Once we finish adding fixtures and tests for other blocks, giving the code base another sweep to remove these might pay dividends. I see we have a few more blocks where these comments were left in place.

File: projects/plugins/jetpack/extensions/blocks/google-calendar/edit.js (hunk 2 above)
Comment (aaronrobertshaw) [reply projects/plugins/jetpack/extensions/blocks/google-calendar/edit.js]: Done. Thanks.

File: projects/plugins/jetpack/extensions/blocks/google-calendar/controls.js (hunk 1 above)
Comment (aaronrobertshaw) [reply projects/plugins/jetpack/extensions/blocks/google-calendar/controls.js]: Thanks. The occurrences of `isLarge` have been removed.

**Review by andrewserong — APPROVED** [review]
Thanks for fixing that up! Just did a quick smoke test, and the console warnings are gone 👍
Looks like CI is complaining that the PR isn't up to date, so might need a `git merge origin/master` before this lands.

**Review by jeherve — APPROVED** [review]
This tests well for me. 🚢

**Comment by aaronrobertshaw** [comment]
Deployed: rWP224500

**Pull request merged**